### PR TITLE
Improve numerical gradient and fix tests

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

**Comment by piiswrong** [comment]
@sxjscience

File: src/operator/sequence_mask-inl.h [hunk 1]
```diff
@@ -108,7 +108,7 @@ class SequenceMaskOp : public Operator {
     if (param_.use_sequence_length) {
       Tensor<xpu, 1, DType> indices =
           in_data[seq_mask::kSequenceLength].get<xpu, 1, DType>(s);
-      SequenceMask(data_grad, indices, static_cast<DType>(param_.value));
+      SequenceMask(data_grad, indices, DType(0));
```
Comment (piiswrong) [inline src/operator/sequence_mask-inl.h]: @sbodenstein gradient of sequence mask was wrong. consider rerun your experiments thats using this op

**Comment by sxjscience** [comment]
Thanks very much!

File: src/operator/sequence_mask-inl.h (hunk 1 above)
Comment (sbodenstein) [inline src/operator/sequence_mask-inl.h]: Thanks for catching that!

**Comment by piiswrong** [comment]
@dsqx71 I refactored tests to limit maximum error instead of average error. Bilinear sampler tests now fails randomly on GPU. Please check implementation.

**Pull request merged**